src/components/application_manager/test/commands/mobile/delete_file_test.cc

```diff
@@ -0,0 +1,238 @@
+    MockHMICapabilities;
+
+ACTION_TEMPLATE(SetArgPointer,
+                HAS_1_TEMPLATE_PARAMS(int, k),
+                AND_1_VALUE_PARAMS(vec)) {
+  *std::tr1::get<k>(args) = *vec;
+}
+
+MATCHER_P(CheckMessageResultCode, result_code, "") {
+  return (*arg)[am::strings::msg_params][am::strings::result_code].asInt() ==
+         result_code;
+}
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kCorrelationId = 10u;
+const int32_t kMenuId = 5;
+}  // namespace
+
+class DeleteFileRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessage();
+    command_ = CreateCommand<DeleteFileRequest>(message_);
+    mock_app_ = CreateMockApp();
+  }
+  DeleteFileRequestPtr command_;
+  MessageSharedPtr message_;
+  MockAppPtr mock_app_;
+};
+
+class DeleteFileResponseTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+};
+
+TEST_F(DeleteFileRequestTest, Run_InvalidApp_UNSUCCESS) {
+  MockAppPtr invalid_app;
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(invalid_app));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL));
+  EXPECT_CALL(app_mngr_, get_settings()).Times(0);
+
+  command_->Run();
+}
+
+TEST_F(DeleteFileRequestTest, Run_HMILevelNone_UNSUCCESS) {
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(am::mobile_api::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+  const uint32_t num = 0;
+  EXPECT_CALL(app_mngr_settings_, delete_file_in_none())
+      .WillOnce(ReturnRef(num));
+  // do not trying delete file twice
+  EXPECT_CALL(*mock_app_, delete_file_in_none_count()).WillOnce(Return(1));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(CheckMessageResultCode(mobile_apis::Result::REJECTED),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command_->Run();
+}
+
+TEST_F(DeleteFileRequestTest, Run_ValidFileName_SUCCESS) {
+  const std::string file_name = "test_file.txt";
+  EXPECT_TRUE(file_system::CreateFile(file_name));
+  (*message_)[am::strings::msg_params][am::strings::sync_file_name] = file_name;
+
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(am::mobile_api::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+  const std::string full_file_path = file_system::CurrentWorkingDirectory();
+  EXPECT_CALL(app_mngr_settings_, app_storage_folder())
+      .WillOnce(ReturnRef(full_file_path));
+
+  am::AppFile file;
+  file.file_name = file_name;
+  file.file_type = mobile_apis::FileType::BINARY;
+
+  EXPECT_CALL(*mock_app_, GetFile(_)).WillOnce(Return(&file));
+  EXPECT_CALL(*mock_app_, DeleteFile(_));
```

Comment:
@AByzhynar, sorry. But it isn't my test. I tried check parameter as I understood this test but it's doesn't work. So, I need much more time to find right behavior for it. Unfortunately I have no more time in my current task.